src/CFamily.UnitTests/Rules/CFamilyRuleConfigProviderTests.cs

```diff
@@ -18,233 +18,86 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
 using System.Collections.Generic;
-using System.IO.Abstractions;
 using FluentAssertions;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Moq;
-using Newtonsoft.Json;
 using SonarLint.VisualStudio.CFamily.Helpers.UnitTests;
 using SonarLint.VisualStudio.Core;
-using SonarLint.VisualStudio.Core.Binding;
-using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.Integration;
 
 namespace SonarLint.VisualStudio.CFamily.Rules.UnitTests
 {
     [TestClass]
     public class CFamilyRuleConfigProviderTests
```

Comment:
IRuleSettingsProviderFactory hides now the details between connected mode and standalone mode so there is no need for the tests anymore. I've kept a single integration test that in fact tests the EffectiveRulesConfigCalculator.